### PR TITLE
Extracted code from the `report` project

### PR DESCRIPTION
**Pull request by marcospri**
For:

- https://github.com/hypothesis/report/issues/47

## Testing 

- https://github.com/hypothesis/report/pull/87 Pointing to a local version of this
- Adapt the location https://github.com/hypothesis/report/pull/87#discussion_r1029091274
- And run `tox -e dev --run-command 'python bin/run_sql_task.py  -t report/create_from_scratch'`

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import os
+
+import pytest
+import sqlalchemy
+
+TEST_SETTINGS = {"database_url": os.environ.get("TEST_DATABASE_URL", "sqlite://")}
```
Comment (marcospri) [inline tests/conftest.py:6]: We can change this back to PostgreSQL when we merge: https://github.com/hypothesis/data-tasks/pull/3

File: src/data_tasks/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+"""Classes for reading and executing tasks defined in SQL files."""
+
+from data_tasks.loader import from_dir
```
Comment (jon-betts) [inline src/data_tasks/__init__.py:3]: The file `core.py` in this directory looks like cruft?

File: setup.cfg [hunk 3]
```diff
@@ -19,6 +19,11 @@ packages = find:
 python_requires = >=3.7
 install_requires =
     importlib_metadata;python_version<"3.8."
+    importlib_resources
+    jinja2
+    SQLAlchemy
+    sqlparse
+    tabulate
```
Comment (jon-betts) [inline setup.cfg:26]: If you do `make template` these disappear. I assume there's a cookiecutter include file for these?

File: tests/conftest.py (hunk 1 above)
Comment (jon-betts) [reply tests/conftest.py:6]: Yeah. This is all a bit overkill for SQLite, but it does allow us to do that.

Honestly this is fine, and fast as well.

File: tests/unit/conftest.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+@pytest.fixture
+def db_session(db_engine):
+    """
+    Yield the SQLAlchemy session object.
+
+    We enable fast repeatable database tests by setting up the database only
+    once per session (see :func:`db_engine`) and then wrapping each test
+    function in a transaction that is rolled back.
```
Comment (jon-betts) [inline tests/unit/conftest.py:13]: I think with an in memory SQLite DB, you'd probably be better off creating it from scratch every time. I wouldn't be surprised to find out it's every bit as quick.

**Review by jon-betts — COMMENTED** [review]
I suspect, but can't totally prove this is basically correct. Running the tests in `report` as suggested cause failures as the tasks still load code from `report.sql_tasks.sql_query`.

As far as this PR goes there are a couple of things I spotted:

 * Running `make template` removes some parts of `setup.cfg`
 * Some random cruft in `data_tasks/core.py`

**Pull request merged**